bip-0352.mediawiki

```diff
@@ -0,0 +1,487 @@
+
+''' P2TR '''
+
+'' Keypath spend ''
+
+    witness:      <signature>
+    scriptSig:    (empty)
+    scriptPubKey: 0 <32-byte-x-only-key>
+                  (0x5120{32-byte-x-only-key})
+
+The sender uses the private key corresponding to the taproot output key (i.e. the tweaked private key). This can be a single private key or an aggregate key (e.g. taproot outputs using MuSig or FROST)<ref name="musig_frost_support">'''Are key aggregation techniques like FROST and MuSig supported?''' While we do not recommend it due to lack of a security proof (except if all participants are trusted or are the same entity), any taproot output able to do a key path theoretically is supported. Any offline key aggregation technique can be used, such as FROST or MuSig. This would require participants to perform the ECDH step collaboratively e.g. ''ECDH = a<sub>1</sub>·B<sub>scan</sub> + a<sub>2</sub>·B<sub>scan</sub> + ... + a<sub>t</sub>·B<sub>scan</sub>'' and ''P = B<sub>spend</sub> + hash(input_hash·ECDH || 0)·G''. Additionally, it may be necessary for the participants to provide a DLEQ proof to ensure they are not acting maliciously.</ref>. The receiver obtains the public key from the ''scriptPubKey'' (i.e. the taproot output key).
+
+'' Script path spend ''
+
+    witness:      <optional witness items> <leaf script> <control block>
+    scriptSig:    (empty)
+    scriptPubKey: 0 <32-byte-x-only-key>
+                  (0x5120{32-byte-x-only-key})
+
+Same as a keypath spend, the sender MUST use the private key corresponding to the taproot output key. If this key is not available, the output cannot be included as an input to the transaction. Same as a keypath spend, the receiver obtains the public key from the ''scriptPubKey'' (i.e. the taproot output key)<ref name="why_always_output_pubkey">''' Why not skip all taproot script path spends? ''' This causes malleability issues for CoinJoins. If the silent payments protocol skipped taproot script path spends, this would allow an attacker to join a CoinJoin round, participate in deriving the silent payment address using the tweaked private key for a key path spend, and then broadcast their own version of the transaction using the script path spend. If the receiver were to only consider key path spends, they would skip the attacker's script path spend input when deriving the shared secret and not be able to find the funds. Additionally, there may be scenarios where the can perform ECDH with the key path private key but spends the output using the script path.</ref>.
```

Comment:
@josibake The language is a bit confusing to me. It says "the sender MUST use the private key corresponding to the taproot output key" Doesn't this mean that `script-path` spends that don't use the taproot tweaked key should be ignored?

Reply:
This is what is explained in the footnote. Imagine you want to send a silent payment to someone, but you're doing so in a coinjoin (i.e. a transaction where other people are allowed to add inputs). One person uses taproot for one of their inputs, then claims they will use the taproot output key (meaning their key will be used for generating the silent payment output). Now as soon as the transaction is finalized, this person swaps out the key path for a script path (this is possible because input scripts are malleable) and this is what gets confirmed on-chain instead. If we followed your suggestion and ignored the script path spend, the silent payment would fail and the money would be lost. This is why we always use the taproot output key for silent payments, regardless of whether it's a script path spend or not.

In other words, we deliberately separated the choice of how taproot outputs get spent from how the silent payment output is generated to work around malleability issues.